cdap-common/src/main/java/co/cask/cdap/common/twill/AbstractDistributedMasterServiceManager.java

```diff
@@ -179,4 +179,41 @@ protected final HttpResponseStatus checkGetStatus(String url) throws Exception {
       }
     }
   }
+
+  @Override
+  public boolean restartAllInstances() {
+    Preconditions.checkArgument(serviceName != null);
+    try {
+      Iterable<TwillController> twillControllerList = twillRunnerService.lookup(Constants.Service.MASTER_SERVICES);
+      if (twillControllerList != null) {
+        for (TwillController twillController : twillControllerList) {
+          // Call restart instances
+          twillController.restartAllInstances(serviceName).get();
+        }
+      }
+    } catch (Throwable t) {
+      LOG.error("Could not change service instance of {} : {}", serviceName, t.getMessage(), t);
+      return false;
+    }
+    return true;
+  }
+
+  @Override
+  public boolean restartInstance(int instanceId, int... moreInstanceIds) {
```

Comment:
I think this should be changed, but probably in another PR.. seems bad to log exception here instead of propagating so that the caller knows what happened.